src/ValueObjects/Session/FormEntries.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Give\ValueObjects\Session;
+
+use Give\Helpers\ArrayDataSet;
+use Give\ValueObjects\ValueObjects;
+use InvalidArgumentException;
+
+class FormEntries implements ValueObjects {
```

Comment:
I like what you're doing here, but these aren't actually Value Objects. a Value Object is a pattern for when an object represents a _single_ value. For example, you may have a `Currency` class which represents a single integer value. This is really just a run-of-the-mill class. Your `ValueObjects` class can be renamed to something like `ArraySerializable` and have `fromArray` and `toArray` methods.

Reply:
@JasonTheAdams Agree with you. `ValueObjects`'s structure must be defined and immutable that means we can not overwrite them.  I find out that there is not any scope of these objects outside or session that why I am moving `FormEntry` and `Donation`to `Session\Objects`.

ref: https://github.com/impress-org/givewp/compare/2a7b11961de2...4bdd65fe7f79

Reply:
resolved